backend/server.py

```diff
@@ -226,43 +235,53 @@ def add_status():
     status_data = {}
     status_data['nodes'] = []
 
-
     return jsonify(return_data)
 
 @app.route('/fetcher/statistics', methods=['GET', 'POST'])
 def add_statistics():
+    timeout=30
+    waiter_statistics.close()
 
     global statistics_data
 
-    while statistics_data is None:
-        pass
+    if statistics_data is None:

```

Comment:
The `if` check for `statistics_data` is `None` is not necessary, as we already handle whether the data has been received through the waiter.